tony-history-server/app/hadoop/Security.java

```diff
@@ -0,0 +1,44 @@
+package hadoop;
+
+import com.typesafe.config.Config;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.HdfsConfiguration;
+import org.apache.hadoop.security.UserGroupInformation;
+
+import java.io.File;
+import java.io.IOException;
+import org.apache.hadoop.yarn.api.ApplicationConstants;
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+import play.Logger;
+
+
+/**
+ * The class handles authentication when cluster is security enabled
+ */
+@Singleton
+public class Security {
+  private static final Logger.ALogger LOG = Logger.of(Security.class);
+  private Config appConf;
+
+  @Inject
+  public Security(Config appConf) {
+    this.appConf = appConf;
+    setUpKeytab(Configuration.getHdfsConf());
+  }
+
+  private void setUpKeytab(HdfsConfiguration hdfsConf) {
+    boolean isSecurityEnabled = hdfsConf.get("hadoop.security.authentication").equals("kerberos");
```

Comment:
You can import and use `org.apache.hadoop.fs.CommonConfigurationKeysPublic.HADOOP_SECURITY_AUTHENTICATION` here instead.

I think you can also use `org.apache.hadoop.security.UserGroupInformation.AuthenticationMethod.KERBEROS.name()`

Reply:
For the first one, it just returns the string "hadoop.security.authentication". The second one, it always returns the string "KERBEROS", regardless of whether or not the Hadoop cluster has Kerberos set up. I don't think these attributes are what we want.

Reply:
I think there's a misunderstanding. What I'm suggesting is this:

```
import static org.apache.hadoop.fs.CommonConfigurationKeysPublic.HADOOP_SECURITY_AUTHENTICATION;
import static org.apache.hadoop.security.UserGroupInformation.AuthenticationMethod.KERBEROS;
...
boolean isSecurityEnabled = hdfsConf.get(HADOOP_SECURITY_AUTHENTICATION).equals(KERBEROS.name());
```

This avoid hardcoding the string literals "hadoop.security.authentication" and "kerberos". If Hadoop ever changes the values of these string literals, we should get the changes for free when we upgrade Hadoop by relying on the string constants Hadoop defines.